lib/dradis/plugins/persistant_permissions.rb

```diff
@@ -0,0 +1,42 @@
+module Dradis
+  module Plugins
+    module PersistantPermissions
+      extend ActiveSupport::Concern
+
+      def update
+        @user = User.find(params[:id])
+
+        Permission.transaction do
+          Permission.where(component: self.class.engine_name, user_id: params[:id]).destroy_all
+
+          params[:permissions]&.each do |permission|
+            # Validate the permission being created is a valid value
+            next unless "::Dradis::Pro::Plugins::#{self.class.engine_name.to_s.classify}::PERMISSIONS".constantize.include?(permission)
```

Comment:
Hmm.. I don't really like this. It's too reliant on the `PERMISSIONS` constant to be defined under `Dradis::Pro::Plugins` namespace. 

Maybe we can pass it inside the permissible_engine method

```ruby
def permissible_engine(engine/plugin_name, permissions)
end
```

Reply:
Sounds like a plan.

Reply:
I was actually thinking maybe to increase the flexibility of the pattern, we replace the check with a method `validate_permissions`

Where the controller can do whatever it wants. That way if at anypoint some kind of new validation is introduced changes won't be needed across the entire plugin.

Reply:
`validate_permissions` sounds good.

Reply:
ohhhh you might like the new solution 😄